### PR TITLE
fix: wrap response with JSONRPCResponse

### PR DESCRIPTION
**Pull request by ernieyang09**
## Summary

Api response should put in JSONRPCResponse result field.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [ ] Refactor (no functionality changes)

## Checklist

- [x] Remove unnecessary console or comment
- [ ] Add unit tests

### CODE REVIEW

File: src/api/JsonRpc.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  result: TResult;
+  error?: undefined;
+}
+
+export interface JSONRPCErrorResponse {
+  jsonrpc: JSONRPC;
+  id: JSONRPCID;
+  result?: undefined;
+  error: JSONRPCError;
+}
```
Comment (wuleoeluw) [inline src/api/JsonRpc.ts]: 1. plz move `interface` or `type` into https://github.com/frigatebird-studio/tonfura-sdk/blob/21068ad99640bfd69b22d1a7017032a354c09777/src/types/TonfuraSDK.d.ts
2. plz correct all `any`

**Comment by wuleoeluw** [comment]
1. plz attach integration testing result, thx
2. plz update docs(`yarn docs && yarn docs-md`) as well

File: src/types/JSONRPC.d.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    jsonrpc: JSONRPC;
+    id: JSONRPCID;
+    result?: undefined;
+    error: JSONRPCError;
+  }
+
+  type JSONRPCResponse<TResponse = any> =
+    | JSONRPCSuccessResponse<TResponse>
+    | JSONRPCErrorResponse;
+}
```
Comment (wuleoeluw) [inline src/types/JSONRPC.d.ts:33]: Can u try ur best to replace `any`

File: src/types/JSONRPC.d.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+declare namespace JSONRPC {
+  type JSONRPC = '2.0';
```
Comment (wuleoeluw) [inline src/types/JSONRPC.d.ts]: rename with `version` might be more suitable

File: src/types/JSONRPC.d.ts (hunk 2 above)
Comment (wuleoeluw) [reply src/types/JSONRPC.d.ts:33]: There already has namespace(prefix) `JSONRPC`
No need to put prefix `JSONRPC` in the name of every type

File: src/types/JSONRPC.d.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  type JSONRPCID = string | number | null;
```
Comment (wuleoeluw) [inline src/types/JSONRPC.d.ts]: rename with `id` might be more suitable

File: src/types/JSONRPC.d.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  type JSONRPCParams<T = any> = T;
```
Comment (wuleoeluw) [inline src/types/JSONRPC.d.ts]: rename with `Params` might be more suitable

File: src/types/JSONRPC.d.ts (hunk 2 above)
Comment (wuleoeluw) [reply src/types/JSONRPC.d.ts:33]: `d.ts` naming reference
https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html

File: src/types/JSONRPC.d.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    method: string;
+    params?: TParams;
+    id?: ID;
+  }
+
+  interface SuccessResponse<TResult> {
+    jsonrpc: VERSION;
+    id: ID;
+    result: TResult;
+    error?: undefined;
```
Comment (wuleoeluw) [inline src/types/JSONRPC.d.ts]: `error` might not exist or be `undefined`
Is this what u want?

File: src/types/JSONRPC.d.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  }
+
+  interface ErrorResponse {
+    jsonrpc: VERSION;
+    id: ID;
+    result?: undefined;
```
Comment (wuleoeluw) [inline src/types/JSONRPC.d.ts]: `result` might not exist or be `undefined`
Is this what u want?

File: src/constants/settings.ts [hunk 8]
```diff
@@ -11,4 +11,4 @@ export const DEFAULT_TONFURA_API_KEY = 'DEFAULT_API_KEY';
 export const DEFAULT_NETWORK = Network.Mainnet;
 export const DEFAULT_MAX_RETRIES = 5;
 
-export const JSON_RPC_VERSION = '2.0';
+export const JSONRPC: JSONRPC.VERSION = '2.0';
```
Comment (wuleoeluw) [inline src/constants/settings.ts]: It should be `JSON_RPC`
ref: https://en.wikipedia.org/wiki/JSON-RPC

File: src/types/TonfuraSDK.d.ts [hunk 9]
```diff
@@ -34,14 +34,63 @@ declare namespace TonfuraSDK {
...
+    | Core.LookupBlock.Response
+    | Core.PackAddress.Response
+    | Core.RunGetMethod.Response
+    | Core.Shards.Response
+    | Core.TryLocateResultTx.Response
+    | Core.TryLocateSourceTx.Response
+    | Core.TryLocateTx.Response
+    | Core.UnpackAddress.Response
+    | Transact.SendBoc.Response
+    | Transact.SendBocReturnHash.Response;
```
Comment (wuleoeluw) [inline src/types/TonfuraSDK.d.ts:91]: Are there any better way to define `Params` and `Response`

**Review by fmpojsz — APPROVED** [review]

**Pull request merged**